content/general/concepts/algorithm/terms/merge-sort/merge-sort.md

```diff
@@ -0,0 +1,141 @@
+---
+Title: 'Merge Sort Algorithm' 
+
+Description: 'Returns a sorted array using divide and conquer.' 
+
+Subjects: 
+  - 'Computer Science'
+  - 'Interview Prep'
+  - 'Code Foundations'
+Tags:
+  - 'Sorting Algorithms'
+  - 'Algorithms'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Merge sort** is a divide-and-conquer sorting [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that breaks down an [array](https://www.codecademy.com/resources/docs/java/arrays) into smaller arrays, sorts them, and then combines the subarrays back together to return a sorted array. 
+
+
+
+## MergeSort Method
+
+Divide:
+  - Base Case: If the input array `data` has only one element, it's already sorted, so return.
+  - Split: Divide the array into two halves, `left` and `right`, of approximately equal size.
+
+Conquer:
+  
+  - [Recursive](https://www.codecademy.com/learn/java-algorithms/modules/recursion-apcs/cheatsheet) Calls: Repeatedly call `mergeSort` on each half, `(left)` and `(right)`, to create subarrays so they are sorted independently.
+
+Combine:
+  
+  - Merge: Call the `merge` function to merge the two sorted halves (each initially of length 1) `left` and `right`, and combine them back into the original array `data` in a sorted manner.
+
+
+### Implementation Pt.1 
+
```

Comment:
give a basic description of the code snippet down below which can give a rough idea for the readers @JustinhSE

Reply:
Just did! @dakshdeepHERE